modules/kafka/kafka-dest-worker.c

```diff
@@ -61,32 +62,57 @@ _publish_message(KafkaDestWorker *self, LogMessage *msg)
 {
   KafkaDestDriver *owner = (KafkaDestDriver *) self->super.owner;
   int block_flag = _is_poller_thread(self) ? 0 : RD_KAFKA_MSG_F_BLOCK;
+  rd_kafka_topic_t *topic;
+  ScratchBuffersMarker marker;
+  GString *topic_name_new = scratch_buffers_alloc_and_mark(&marker);
```

Comment:
if you are not doing lots of allocations in this (and child) functions, you can just get away by using scratch_buffers_alloc() (and not using reclaim at exit points), as the thread function in LogThreadedDestination would call the garbage collector after every call to this function:

https://github.com/syslog-ng/syslog-ng/blob/edfe167e3f76cb419c84bea09ca3ca42d6da84f4/lib/logthrdest/logthrdestdrv.c#L415-L418